### ISSUE TITLE
Failed to create output folder

### ISSUE THREAD

**Issue by bhsimon0810** [sskodje/ScreenRecorderLib#37]
Hi, I met this problem when I tried to run WPF demo. When I debug the demo, the _StatusTextBlock.Text_ changed from _IDLE_ to _ERROR_, and the error is **"failed to create output folder"**. Could you show some instructions on how to solve this problem? Thanks

**Comment by sskodje** [comment]
Sounds like a problem with write access. Does it happen if you run it as administrator?

**Comment by FoxPride** [comment]
I had same issue and run as administrator doesn't help. 
My case: 
1) if I run program in debug mode from Visual studio then everything is OK and I can record video
2) if I run program from debug\release folder directly then I get error from "utilities.h -> static bool  CreateAllDirs(std::wstring path) -> BOOL result = CreateDirectory(path.substr(0, pos).c_str(), nullptr);" BUT if I use **record to stream** checkbox then everything is OK again
3) if I deploy program with wix installer then install it then everything is OK again...
Once I faced this error in Visual studio debugging mode. I tried different settings for project and when I changed Windows SDK version back and forth then magically error disappeared and now I have only case №2 left.
The thing is that most of our users not facing this error but around 1 from 10000 gets it and I have no idea why.

**Comment by sskodje** [comment]
This is really weird. I don't understand why Directory.CreateDirectory from c# and CreateDirectory from c++ should behave differently. I pushed commit 3056ea7440ee96de08a8e5238badd33079af475e where i replaced CreateDirectory with std::filesystem::create_directories, and also made it print the error message in the error return. Please test it and see if it fixes your issue.

**Comment by FoxPride** [comment]
Just tested it, works like a charm in any case from above. 
One note: std::filesystem::create_directories(directory, ec) returns false if directory already exists so I change code a bit to
`	std::error_code ec;
	if (std::filesystem::exists(directory))
	{
		LOG(L"output folder is already exists");
		m_OutputFolder = directory;
	}
	else
	{
		if (std::filesystem::create_directories(directory, ec))
		{
			LOG(L"output folder is ready");
			m_OutputFolder = directory;
		}
		else
		{
			// Failed to create directory.
			ERR(L"failed to create output folder");
			if (RecordingFailedCallback != nullptr)
				RecordingFailedCallback(L"Failed to create output folder: " + utilities::s2ws(ec.message()));
			return E_FAIL;
		}
	}`

**Comment by FoxPride** [comment]
```cs
	std::error_code ec;
	if (std::filesystem::exists(directory))
	{
		LOG(L"output folder is already exists");
		m_OutputFolder = directory;
	}
	else
	{
		if (std::filesystem::create_directories(directory, ec))
		{
			LOG(L"output folder is ready");
			m_OutputFolder = directory;
		}
		else
		{
			// Failed to create directory.
			ERR(L"failed to create output folder");
			if (RecordingFailedCallback != nullptr)
				RecordingFailedCallback(L"Failed to create output folder: " + utilities::s2ws(ec.message()));
			return E_FAIL;
		}
	}
	```
![изображение](https://user-images.githubusercontent.com/30706733/68393631-7abd7d80-017d-11ea-92f2-9ec4c2fce54d.png)

**Comment by sskodje** [comment]
Ah.. Thanks for the heads up. I'll push a new release with the changes then.

**Comment by sskodje** [comment]
Fixed with c37cd96650bcac75a579deda124f2ec0ad1ac05b
